### PR TITLE
fix: dev environment git auth env vars

### PR DESCRIPTION
**Pull request by macox**
Signed-off-by: Mark Cox <markcox20@hotmail.com>

#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

Make the gitops dev environment pipeline use the git values of the bot user.

#4943

### CODE REVIEW

File: pkg/cmd/step/verify/step_verify_environments.go [hunk 1]
```diff
@@ -180,6 +187,45 @@ func (o *StepVerifyEnvironmentsOptions) prDevEnvironment(gitRepoName string, env
...
+		}
+		envVars = append(envVars, gitConfig...)
+		projectConf.PipelineConfig.Pipelines.Release.Pipeline.Environment = envVars
+
+		err = projectConf.SaveConfig(fileName)
+		if err != nil {
+			return errors.Wrapf(err, "failed to write to %s", fileName)
+		}
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (macox) [reply pkg/cmd/step/verify/step_verify_environments.go]: Meaning if the dev environment is already created? true, maybe I should also set here https://github.com/jenkins-x/jx/blob/master/pkg/cmd/step/git/step_git_validate.go#L55

File: pkg/cmd/step/verify/step_verify_environments.go [hunk 2]
```diff
@@ -180,6 +187,45 @@ func (o *StepVerifyEnvironmentsOptions) prDevEnvironment(gitRepoName string, env
...
+	if username != "" && email != "" {
+		fileName := filepath.Join(dir, config.ProjectConfigFileName)
+		projectConf, err := config.LoadProjectConfigFile(fileName)
+		if err != nil {
+			return errors.Wrapf(err, "failed to load project config file %s", fileName)
+		}
+		gitConfig := []corev1.EnvVar{
+			{
+				Name:  "GIT_AUTHOR_NAME",
+				Value: username,
```
[the remark this answers is not in the document]
Comment (macox) [reply pkg/cmd/step/verify/step_verify_environments.go]: Is was being set as username here https://github.com/jenkins-x/jx/blob/master/pkg/cmd/step/git/step_git_validate.go#L55 ?

File: pkg/cmd/step/verify/step_verify_environments.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmuir) [reply pkg/cmd/step/verify/step_verify_environments.go]: I mean the env vars are set when the pipeline is created. This happens inside the pipeline. So you won't see these env vars being used until the next time the pipeline starts.

**Pull request merged**